### PR TITLE
Feature/dask

### PR DESCRIPTION
**Pull request by gsaluja9**

### CODE REVIEW

File: aperturedb/DaskManager.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import multiprocessing as mp
+
+from aperturedb.Stats import Stats
+
+logger = logging.getLogger(__name__)
+
+
+class DaskManager:
+    def __init__(self, num_workers: int = -1):
```
Comment (drewaogle) [inline aperturedb/DaskManager.py:18]: what does -1 mean for num_workers, controlled by dask?

File: examples/dask/ingest_ssh.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from dask import dataframe
+import os
+from aperturedb import EntityDataCSV
+from dask.distributed import Client, SSHCluster, performance_report
+
+if __name__ == "__main__":
+    batchsize = 4000
+    numthreads = 14
+
+    # cluster = SSHCluster(
```
Comment (drewaogle) [inline examples/dask/ingest_ssh.py:11]: should this be an optional argument or have a comment?

**Review by drewaogle — COMMENTED** [review]
looks very good!

File: aperturedb/CSVParser.py [hunk 3]
```diff
@@ -15,18 +19,30 @@ class CSVParser(Subscriptable):
...
-
-        self.validate()
+    def __init__(self, filename, df=None, use_dask=False):
+        self.use_dask = use_dask
+        self.filename = filename
+
+        if not use_dask:
+            if df is None:
+                self.df = pd.read_csv(filename)
+            else:
```
[the remark this answers is not in the document]
Comment (gsaluja9) [reply aperturedb/CSVParser.py:29]: yes. Let me add some comments about some of the changes.

File: examples/dask/ingest_ssh.py (hunk 2 above)
Comment (gsaluja9) [reply examples/dask/ingest_ssh.py:11]: Oh this was an unfinished experiment. I'll remove this file for now.

**Review by drewaogle — APPROVED** [review]
looks good!

File: test/conftest.py [hunk 4]
```diff
@@ -38,15 +48,18 @@ def insert_data_from_csv(in_csv_file, rec_count=-1):
...
+        use_dask = False
+        if hasattr(request, "param"):
+            use_dask = request.param
+        data = file_data_pair[in_csv_file](in_csv_file, use_dask=use_dask)
         if rec_count != -1:
             data = data[:rec_count]
 
         loader = ParallelLoader(db)
         loader.ingest(data, batchsize=99,
-                      numthreads=31,
```
Comment (luisremis) [inline test/conftest.py:48]: I would revert this change, so that we tests with stranger number of threads. This used to catch certain weird sitautions
